xayn-ai/src/coi/config.rs

```diff
@@ -207,3 +208,61 @@ impl Config {
         self.kpe.penalty.len()
     }
 }
+
+mod serde_horizon_days {
+    use std::time::Duration;
+
+    use serde::{Deserialize, Deserializer, Serialize, Serializer};
+
+    use super::SECONDS_PER_DAY;
+
+    pub(super) fn serialize<S>(horizon: &Duration, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        (horizon.as_secs() / SECONDS_PER_DAY).serialize(serializer)
+    }
+
+    pub(super) fn deserialize<'de, D>(deserializer: D) -> Result<Duration, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        u64::deserialize(deserializer).map(|days| Duration::from_secs(SECONDS_PER_DAY * days))
+    }
+
+    #[cfg(test)]
+    mod tests {
+        use std::error::Error;
+
+        use serde_json::{from_str, to_string};
+
+        use super::*;
+
+        #[derive(Deserialize, Serialize)]
+        struct Days(#[serde(with = "super")] Duration);
+
```

Comment:
Personally, I would add at least one test that also checks if the serialized format is what we expect, e.g. serialize `Duration::from_secs(SECONDS_PER_DAY * 30)` to `30` (assert), then deserialize again (assert).